src/helpers/getEntanglementsHashes.ts

```diff
@@ -13,15 +13,44 @@ function parseEntanglementRegistered({
   return transferEventInterface.parseLog({ data, topics })
 }
 
-export default async function getEntanglementsHashes(type: number) {
-  const contract = getKetlAttestationContract()
-
-  const entanglements = await contract.queryFilter(
-    contract.filters.EntanglementRegistered()
+export default async function getEntanglementsHashes(
+  type: number,
+  hash: string,
+  attestation: string
+) {
+  const batchSize = await ketlAttestationContract.minimumEntanglementCounts(
+    type
   )
-
-  return entanglements
+  const entanglements = await ketlAttestationContract.queryFilter(
+    ketlAttestationContract.filters.EntanglementRegistered()
+  )
+  const filteredEntanglements = entanglements
     .map(parseEntanglementRegistered)
     .filter(({ args }) => Number(args[0]) === type)
-    .map(({ args }) => BigNumber.from(args[1]).toHexString())
+    .map(({ args }) => BigNumber.from(args[1]).toHexString().toLowerCase())
+
+  const hashIndex = filteredEntanglements.indexOf(hash)
+
+  const isUsedAttestation = await checkAttestationHash(attestation)
+
+  if (isUsedAttestation && hashIndex === -1)
+    throw new Error(
+      `This verification method has already been used for another account! ${attestation} ${hash} ${filteredEntanglements.join(
+        ', '
+      )}`
+    )
+
+  if (hashIndex === -1) {
+    throw new Error(
+      `This hash isn't among those added to the blockchain  ${attestation} ${hash}`
+    )
+  }
+
+  const batchHashIndex = hashIndex - (hashIndex % batchSize) + batchSize
+
+  if (batchHashIndex > filteredEntanglements.length) {
+    throw new Error(`Batch hasn't been completed for this account yet`)
```

Comment:
Why we may want to throw and error when trying to `get` entanglements?

Reply:
If the batch is not completed, then it is better to throw an error otherwise the user will generate an incorrect proof